permission_handler_web/lib/web_delegate.dart

```diff
@@ -0,0 +1,224 @@
+          audioTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestCameraPermission(html.MediaDevices devices) async {
+    html.MediaStream? mediaStream;
+
+    try {
+      mediaStream = await devices.getUserMedia({'video': true});
+
+      // In browsers, calling [getUserMedia] will start the recording
+      // automatically right after. This is undesired behavior as
+      // [requestPermission] is expected to request permission only.
+      //
+      // The manual stop action is then needed here for to stop the automatic
+      // recording.
+
+      if (mediaStream.active!) {
+        final videoTracks = mediaStream.getVideoTracks();
+        if (videoTracks.isNotEmpty) {
+          videoTracks[0].stop();
+        }
+      }
+    } on html.DomException {
+      return false;
+    }
+
+    return true;
+  }
+
+  Future<bool> _requestNotificationPermission() async {
+    bool granted = false;
+    html.Notification.requestPermission().then((permission) => {
+          if (permission == "granted") {granted = true}
+        });
+
+    return granted;
+  }
+
+  Future<bool> _requestLocationPermission(html.Geolocation geolocation) async {
+    try {
+      await geolocation.getCurrentPosition();
+      return true;
+    } on html.PositionError {
+      return false;
+    }
+  }
+
+  Future<PermissionStatus> _requestSingularPermission(
+      Permission permission) async {
+    bool permissionGranted = false;
+
+    switch (permission) {
+      case Permission.microphone:
+        permissionGranted = await _requestMicrophonePermission(devices!);
+        break;
+      case Permission.camera:
+        permissionGranted = await _requestCameraPermission(devices!);
+        break;
+      case Permission.notification:
+        permissionGranted = await _requestNotificationPermission();
+        break;
+      case Permission.location:
+        permissionGranted = await _requestLocationPermission(geolocation!);
+        break;
+      default:
+        throw UnimplementedError(
+          '_requestSingularPermission() has not been implemented for '
+          '${permission.toString()} on web.',
+        );
+    }
+
+    if (!permissionGranted) {
+      return PermissionStatus.permanentlyDenied;
+    }
+    return PermissionStatus.granted;
+  }
+
+  /// Requests the user for access to the supplied list of [Permission]s, if
+  /// they have not already been granted before.
+  ///
+  /// Returns a [Map] containing the status per requested [Permission].
+  Future<Map<Permission, PermissionStatus>> requestPermissions(
+      List<Permission> permissions) async {
+    final Map<Permission, PermissionStatus> permissionStatusMap = {};
+
+    for (final permission in permissions) {
+      try {
+        permissionStatusMap[permission] =
+            await _requestSingularPermission(permission);
+      } on UnimplementedError {
+        continue;
+      }
```

Comment:
Let's not swallow the exception but let consumers know they are requesting permissions that are not supported.

Reply:
done.